### ISSUE TITLE
Drag to reposition panel

### ISSUE THREAD

**Issue by meandavejustice** [meandavejustice/min-vid#74]
refs #2 and #3

**Comment by meandavejustice** [comment]
Latest proposal for this from @johngruen was to do the following:
* on `mousedown` resize panel to fill browser window with transparent container element
* make video element draggable within the panel container [react-dnd](https://github.com/gaearon/react-dnd) | [example](http://gaearon.github.io/react-dnd/examples-drag-around-naive.html)
* drag to desired destination
* on `mouseup` send new panel position to main addon thread and resize panel container

**Comment by 6a68** [comment]
@kaganjd and I are going to work through this next

**Comment by meandavejustice** [comment]
mockups: https://mozilla.invisionapp.com/share/ZF7K069R9#/screens/163789331

**Comment by 6a68** [comment]
I think I've figured out how to get the panel inside a container element.

It [looks like](https://dxr.mozilla.org/mozilla-central/source/addon-sdk/source/lib/sdk/panel.js#264) the SDK's `panel.show()` method actually takes two undocumented arguments.

You can call `panel.show(options, anchorElement)` where
- `options` is a config object containing position, width, defaultWidth, etc. This config object is merged into the options that were passed into the panel constructor.
- `anchorElement` seems to be exactly what we need: a way to specify a specific XUL element that could serve as a wrapper / resizable drop target.

Even better, if you call `panel.show(anchorElement)`, the function will rearrange the arguments internally, so there's no need to pass in an empty `options` argument if none is needed.

Passing in an `anchorElement` is [marked as deprecated](https://dxr.mozilla.org/mozilla-central/source/addon-sdk/source/lib/sdk/panel.js#269-273), but I'm pretty sure the SDK is not being actively developed while WebExtensions is the focus, so I don't think this function is likely to change.

**Comment by meandavejustice** [comment]
@6a68 I think what you are describing is a bit more complicated than what I had invisioned. I thought we could just resize the panel to the browser viewport width and height, set the background to transparent, and drag within the document. So everything will still be done as if it were in a normal html document

**Comment by meandavejustice** [comment]
@6a68 I made a really buggy version of what I am talking about here http://davejustice.com/shitty-drag-drop-demo/ using https://github.com/mzabriskie/react-draggable I think we should avoid https://github.com/gaearon/react-dnd as it appears to be one of the most overengineered libraries I've seen.

**Comment by 6a68** [comment]
Yeah, I can't figure out how to make the XUL container element transparent and sized to cover the entire window. I could push further, but it might take a month, or it might never work. Seems better to resize the panel itself and drag the video around inside the page :-)

**Comment by meandavejustice** [comment]
@6a68 in case that we rezize the panel, we will still need to make the background transparent so it appears that the panel is being resized. Which element has the background that cannot be set to transparent?

**Comment by 6a68** [comment]
whew, that was a workout!

figured out how to make the panel transparent. The trick is styling the anonymous content inserted by XBL.

Note that html and body need to have their backgrounds set to transparent in style.css.

Other than that one-liner CSS change, this JS seems to do it for me (should be inserted anywhere in index.js)

```js
function makePanelTransparent() {
  // Get the panel element in the XUL DOM and make its background transparent.
  // TODO: not sure this is e10s compatible.
  const el = getActiveView(panel);
  el.style.background = 'rgba(0,0,0,0)';

  // Go up the XUL DOM till you hit the Document (nodeType 9).
  let parentNode = el;
  while (parentNode !== null && parentNode.nodeType !== 9) {
    parentNode = parentNode.parentNode;
  }

  if (!parentNode) {
    console.error('unable to find the document parent; giving up');
    return;
  }

  // Now that we've found it, call the document a document.
  const xulDocument = parentNode;

  // Use the document pointer to access and style 'anonymous' content.
  const xulContainer = xulDocument.getAnonymousElementByAttribute(el, 'class', 'panel-arrowcontent')
  xulContainer.style.background = 'rgba(0,0,0,0)';
  xulContainer.style.boxShadow = 'none';
}
makePanelTransparent();
```

**Comment by 6a68** [comment]
@meandavejustice ok, unstealing. let's sync up when you're back though

**Comment by meandavejustice** [comment]
fixed in #132
